### PR TITLE
feat: setup boilerplates

### PR DESCRIPTION
**Pull request by FabricioBattaglia**
Reworked boilerplates made during meeting.
Created routes that should be close to the actual routes that will be used
Organized folders

### CODE REVIEW

File: controllers/transactionController.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	TranType    string
+	TranDate    time.Time
+	Description string
+}
+
+func TransactionCreate(c *gin.Context) {
+	//get date from request body
+	//create user
+	//return it
+	//GORM create guide
```
Comment (victormoraesgs) [inline controllers/transactionController.go:22]:
```suggestion
```
No need for these comments

File: controllers/transactionController.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		return
+	}
+
+	c.JSON(200, gin.H{
+		"transaction": transaction,
+	})
+}
+
+func TransactionFetch(c *gin.Context) {
+	//get user
```
Comment (victormoraesgs) [inline controllers/transactionController.go:43]:
```suggestion
```

File: controllers/transactionController.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	var transactions []models.Transaction
+
+	initializers.DB.Find(&transactions)
+
+	//respond
```
Comment (victormoraesgs) [inline controllers/transactionController.go:48]:
```suggestion
```

File: controllers/userController.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	"github.com/Today-budget-app/backend/models"
+	"github.com/gin-gonic/gin"
+)
+
+func UserCreate(c *gin.Context) {
+	//get date from request body
+	//create user
+	//return it
+	//GORM create guide
+
```
Comment (victormoraesgs) [inline controllers/userController.go:14]:
```suggestion
```

File: controllers/userController.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		return
+	}
+
+	c.JSON(200, gin.H{
+		"user": user,
+	})
+}
+
+func UserFetch(c *gin.Context) {
+	//get user
```
Comment (victormoraesgs) [inline controllers/userController.go:37]:
```suggestion
```

File: controllers/userController.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	c.Bind(&content)
+	user := models.User{Username: content.Username, Email: content.Email}
+	result := initializers.DB.Find(&user)
+
+	if result.Error != nil {
+		c.Status(404)
+		return
+	}
+
+	//respond
```
Comment (victormoraesgs) [inline controllers/userController.go:52]:
```suggestion
```

**Review by victormoraesgs — CHANGES_REQUESTED** [review]

**Comment by FabricioBattaglia** [comment]
Followed suggestions, changes made.

**Review by victormoraesgs — APPROVED** [review]
LGTM

**Pull request merged**